assets/js/app.js

```diff
@@ -0,0 +1,288 @@
+            return await response.json();
+        } catch (error) {
+            console.error("Error fetching data:", error);
+            throw error;
+        }
+    };
+
+    // Function to display the coins cards on the page
+    const renderCoins = async () => {
+        try {
+            let coins = await getData("https://api.coingecko.com/api/v3/coins/list");
+            coins = coins.slice(0, 100);
+
+            const selectedCoins = loadSelectedCoins();
+
+            const html = coins
+                .map(
+                    (coin) => `
+                    <div class="card coin-card" style="width: 18rem;" data-coin-name="${coin.name}">
+                        <div class="card-body">
+                            <h5 class="card-title">${coin.name}</h5>
+                            <p class="card-text"> ${coin.id}</p>
+                            <p class="card-text"></p>
+                            <button class="btn btn-primary more-info-btn" data-coin="${coin.id}">More Info</button>
+                        </div>
+                        <div class="form-check form-switch">
+                            <input class="form-check-input coin-switch" type="checkbox" role="switch" id="switch-${coin.id}">
+                            <label class="form-check-label" for="switch-${coin.id}"></label>
+                        </div>
+                    </div>`
+                )
+                .join("");
+
+            document.getElementById("cards-container").innerHTML = html;
+
+            // Marking the selected currencies
+            selectedCoins.forEach((coinId) => {
+                const switchElement = document.getElementById(`switch-${coinId}`);
+                if (switchElement) {
+                    switchElement.checked = true;
+                }
+            });
+
+            updateSelectedCoinsDisplay();
+
+            // Listens for a click on the "More Info" button on each coin card
+            document.querySelectorAll(".more-info-btn").forEach((button) => {
+                button.addEventListener("click", async (event) => {
+                    const coinId = event.target.dataset.coin;
+                    const coinData = await getData(`https://api.coingecko.com/api/v3/coins/${coinId}`);
+                    showCoinInfo(coinData);
+                });
+            });
+
+            // Handling coins selection with Switch
+            document.querySelectorAll('.coin-switch').forEach(function(switchElement) {
+                switchElement.addEventListener('change', function() {
+                    updateSelectedCoinsDisplay();
+
+                    // Count the number of marked coins
+                    let selectedCoins = document.querySelectorAll('.coin-switch:checked').length;
+
+                    if (selectedCoins > 5) {
+                        switchElement.checked = false; // Uncheck the last selected
+                        alert("You can select up to 5 coins only.");
```

Comment:
here you missed one of the key features of the spec: a modal that allows the user to deselect one of the currently selected coins. an alert is not enough